cooked-validators/src/Cooked/MockChain/Wallet.hs

```diff
@@ -47,15 +50,30 @@ walletPKHashToId = flip M.lookup walletPKHashToIdMap
 walletPK :: Wallet -> Pl.PubKey
 walletPK = Pl.unPaymentPubKey . CW.paymentPubKey
 
+walletStakingPK :: Wallet -> Maybe Pl.PubKey
+walletStakingPK = fmap Crypto.toPublicKey . walletStakingSK
+
 walletPKHash :: Wallet -> Pl.PubKeyHash
 walletPKHash = Pl.pubKeyHash . walletPK
 
+walletStakingPKHash :: Wallet -> Maybe Pl.PubKeyHash
+walletStakingPKHash = fmap Crypto.pubKeyHash . walletStakingPK
+
 walletAddress :: Wallet -> Pl.Address
-walletAddress = (`Pl.Address` Nothing) . Pl.PubKeyCredential . walletPKHash
+walletAddress w =
+  Pl.Address
+    (Pl.PubKeyCredential $ walletPKHash w)
+    (Pl.StakingHash . Pl.PubKeyCredential <$> walletStakingPKHash w)
 
 walletSK :: CW.MockWallet -> Pl.PrivateKey
 walletSK = Pl.unPaymentPrivateKey . CW.paymentPrivateKey
 
+-- Massive hack to be able to open a MockPrivateKey
+newtype HACK = HACK {please :: Crypto.XPrv}
```

Comment:
How does this hack work? This type does not seem to be used elsewhere for now.

Reply:
Added a comment and a test in [bb80f39](https://github.com/tweag/plutus-libs/pull/81/commits/bb80f39cb7720e3d87ac7b9401578800e656a81c):

```haskell
-- Massive hack to be able to open a MockPrivateKey; this is needed because
-- the constructor and accessor to MockPrivateKey are not exported. Hence,
-- we make an isomorphic datatype, unsafeCoerce to this datatype then extract
-- whatever we need from it.
```